### PR TITLE
Improvement: Added heading tag in Taxonomy list block

### PR DESCRIPTION
**Pull request by supriyajjadhav**
[Trello](https://trello.com/c/7uiOsp6P/208-improvement-taxonomy-list-html-tag-option-for-title)

### Description
Added heading tag features for the title.

### Types of changes
- Added titleTag 

### How has this been tested?
Checked whether it is working properly or not on front-end and back-end.

### Checklist:
- [x] My code is tested
- [x] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [] I've included any necessary tests 
- [x] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: dist/blocks/taxonomy-list/class-uagb-taxonomy-list.php [hunk 1]
```diff
@@ -474,7 +481,7 @@ public function list_html( $attributes ) {
 					<ul class="uagb-list-wrap">
 						<?php foreach ( $newcategoriesList as $key => $value ) { ?>
 							<li class="uagb-tax-list">
-								<div class="uagb-tax-link-wrap">
+								<<?php echo esc_html( $titleTag ); ?> class="uagb-tax-link-wrap">
```
Comment (RajkiranBagal) [inline dist/blocks/taxonomy-list/class-uagb-taxonomy-list.php:484]: Please check Backward compatibility of Changing a div to `h` Tag

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

**Comment by RajkiranBagal** [comment]
@supriyajjadhav Resolve conflicts.

File: dist/blocks/taxonomy-list/class-uagb-taxonomy-list.php (hunk 1 above)
Comment (supriyajjadhav) [reply dist/blocks/taxonomy-list/class-uagb-taxonomy-list.php:484]: Checked Backward compatibility,  no issues were found. 
Also added a div tag in the list of titleTag

**Review by RajkiranBagal — APPROVED** [review]

File: dist/blocks/taxonomy-list/class-uagb-taxonomy-list.php (hunk 1 above)
Comment (RajkiranBagal) [reply dist/blocks/taxonomy-list/class-uagb-taxonomy-list.php:484]: @akshaylakh Tested this & found an issue with Backward compatibility which was a simple case.

**Review by sandeshjangam — APPROVED** [review]

**Review by sandeshjangam — APPROVED** [review]

**Pull request merged**